p2p/dial.go

```diff
@@ -388,6 +389,9 @@ func (d *dialScheduler) checkDial(n *enode.Node) error {
 	if n.ID() == d.self {
 		return errSelf
 	}
+	if n.IP() != nil && n.TCP() < 1024 {
```

Comment:
Can `.IP()` be nil here? And if so, shouldn't that return some (other) error?

Reply:
Nodes can have no IP if the user just adds the ID as a static node. Nodes from discovery will always have an IP though.